app/letter_branding/letter_branding_rest.py

```diff
@@ -0,0 +1,84 @@
+from celery import current_app
+from sqlalchemy.exc import IntegrityError
+
+from flask import Blueprint, jsonify, request
+
+from app.dao.letter_branding_dao import (
+    dao_get_all_letter_branding, dao_create_letter_branding,
+    dao_update_letter_branding,
+    dao_get_letter_branding_by_id
+)
+from app.errors import register_errors
+from app.letter_branding.letter_branding_schema import post_letter_branding_schema
+from app.models import LetterBranding
+from app.schema_validation import validate
+
+letter_branding_blueprint = Blueprint('letter_branding', __name__, url_prefix='/letter-branding')
+register_errors(letter_branding_blueprint)
+
+
+@letter_branding_blueprint.errorhandler(IntegrityError)
+def handle_integrity_error(exc):
+    """
+    Handle integrity errors caused by the unique constraint
+    """
+    if 'domain' in str(exc):
+        return jsonify(
+            result='error',
+            message={'name': ["Duplicate domain '{}'".format(
+                exc.params.get('domain')
+            )]}
+        ), 400
+    if 'name' in str(exc):
```

Comment:
i'm going to have a look at what `str(exc)` actually returns because this check is pretty wide-reaching and scares me a little

Reply:
update: changed this (it was erroneously catching `filename` under the `name` param for example)